File: core/src/main/java/hello/core/member/service/MemberServiceImpl.java
package hello.core.member.service;

import hello.core.member.data.Member;
import hello.core.member.repository.MemberRepository;

public class MemberServiceImpl implements MemberService {

    private final MemberRepository memberRepository;

    public  MemberServiceImpl(MemberRepository memberRepository) {
        this.memberRepository = memberRepository;
    }

    @Override
    public void join(Member member) {
        memberRepository.save(member);
    }

    @Override
    public Member findMember(Long memberId) {
        return memberRepository.findById(memberId);
    }

    // Test용
    public MemberRepository getMemberRepository() {
        return memberRepository;
    }

}

File: core-kotlin/src/main/kotlin/hello/corekotlin/member/service/MemberServiceImpl.kt
package hello.corekotlin.member.service

import hello.corekotlin.member.data.Member
import hello.corekotlin.member.repository.MemberRepository

class MemberServiceImpl(
    private val memberRepository: MemberRepository
): MemberService {

    override fun join(member: Member) {
        memberRepository.save(member)
    }

    override fun findMember(memberId: Long): Member {
        return memberRepository.findById(memberId)
    }

    fun getMemberRepository(): MemberRepository {
        return memberRepository;
    }
}